xdsl/transforms/varith_transformations.py

```diff
@@ -209,6 +197,7 @@ def apply(self, ctx: MLContext, op: builtin.ModuleOp) -> None:
                     MergeVarithOpsPattern(),
                 ]
             ),
+            walk_reverse=True,
         ).rewrite_op(op)
 
 
```

Comment:
This change seems to be already in https://github.com/xdslproject/xdsl/pull/3330/files. Maybe rebase after merging #3330.

Reply:
Good spot, I had merged the fix from #3330 into this branch to finish development. I have merged main and the changes have disappeared from this PR's diff.